presto-main/src/main/java/io/prestosql/sql/planner/SymbolsExtractor.java

```diff
@@ -181,5 +191,15 @@ protected Void visitIdentifier(Identifier node, ImmutableSet.Builder<QualifiedNa
             builder.add(QualifiedName.of(node.getValue()));
             return null;
         }
+
+        @Override
+        protected Void visitSubqueryExpression(SubqueryExpression node, ImmutableSet.Builder<QualifiedName> context)
```

Comment:
what if there is a reference to outer column within subquery?

Reply:
That’s taken care of during analysis. See the comment where this is used in join planning.

There’s a broader cleanup that needs to be done at some point. Unfortunately, we currently resolve columns in join planning to determine where they originate. This should be done entirely during analysis.

Reply:
Could we have a test case for such subquery (referencing outer scope)?

Reply:
We already have a bunch of them. See the tests further down below the one I added. E.g.,

```
        assertThatThrownBy(() -> assertions.query(
                "SELECT * FROM (VALUES 1, 2, NULL) t(x) FULL JOIN (VALUES 1, 3, NULL) u(x) ON u.x IN (VALUES t.x)"))
                .hasMessage("line 1:93: Reference to column 't.x' from outer scope not allowed in this context");
```